app/recipe/models.py

```diff
@@ -0,0 +1,30 @@
+from django.db import models
+
+
+class RecipeManager(models.Manager):
+    def create_recipe(self, recipe: dict, ingredients: list):
+        recipe = Recipe.objects.create(**recipe)
+        recipe.add_ingredients(ingredients=ingredients)
+
+        return recipe
+
+
+class Recipe(models.Model):
+    objects = RecipeManager()
+
+    created_at = models.DateTimeField(auto_now_add=True)
+    name = models.CharField(max_length=255)
+    description = models.CharField(max_length=2000)
+
+    def add_ingredients(self, ingredients):
+        for ingredient in ingredients:
+            Ingredient.objects.create(recipe=self, **ingredient)
+
+    def update_ingredients(self, ingredients):
+        Ingredient.objects.filter(recipe=self).delete()
```

Comment:
It depends on the API usage if it is OK to delete and replace components, e.g. the ingredients will have a different primary key (id). If all the API exposes is the `Recipe` then this is probably fine if the client would never see the `Ingredient` id.

Reply:
From the exercice I understand that ingredients are only tight to a recipe and that should be deleted when update. But thanks for pointing this. It's good to keep it in mind! :)